vortex-array/src/arrays/null/compute/cast.rs

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use vortex_dtype::DType;
+use vortex_error::{VortexResult, vortex_bail};
+
+use crate::arrays::{NullArray, NullVTable};
+use crate::compute::{CastKernel, CastKernelAdapter};
+use crate::{ArrayRef, register_kernel};
+
+impl CastKernel for NullVTable {
+    fn cast(&self, array: &NullArray, dtype: &DType) -> VortexResult<Option<ArrayRef>> {
+        // Null can only be cast to Null
+        match dtype {
```

Comment:
I think you can support arbitrary cast here? You just create AllInvalid array of a given dtype matching the length? I wonder what arrow does for these. Maybe we should remove this array and say null arrays don't roundtrip